azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Webhook.java

```diff
@@ -0,0 +1,663 @@
+
+        /**
+         * The stage of the webhook definition allowing to specify the custom headers that will be added to the notifications.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithCustomHeaders<ParentT> {
+            /**
+             * Specifies custom headers that will be added to the notifications.
+             *
+             * @param name of the optional header
+             * @param value of the optional header
+             * @return the next stage of the resource update
+             */
+            WithAttach<ParentT> withCustomHeader(String name, String value);
+
+            /**
+             * Specifies the the service URI for post notifications.
+             *
+             * @param customHeaders the "Name=Value" custom headers.
+             * @return the next stage of the resource update
+             */
+            WithAttach<ParentT> withCustomHeaders(Map<String, String> customHeaders);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the scope of repositories where the event can be triggered.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithRepositoriesScope<ParentT> {
+            /**
+             * Updates the scope of repositories where the event can be triggered.
+             * <p>
+             * For example, 'foo:*' means events for all tags under repository 'foo'. 'foo:bar' means events for 'foo:bar' only.
+             *  'foo' is equivalent to 'foo:latest', empty means all events.
+             *
+             * @param repositoriesScope the scope of repositories where the event can be triggered; empty means all events
+             * @return the next stage of the resource update
+             */
+            WithAttach<ParentT> withRepositoriesScope(String repositoriesScope);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the default status of the webhook after being created.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithDefaultStatus<ParentT> {
+            /**
+             * Updates the default status of the webhook.
+             *
+             * @param defaultStatus indicates whether the webhook is enabled or disabled after the being created
+             * @return the next stage of the resource update
+             */
+            WithAttach<ParentT> withDefaultStatus(WebhookStatus defaultStatus);
+        }
+
+        /**
+         * The stage of the webhook definition allowing to specify the tags.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithOrWithoutTags<ParentT> {
```

Comment:
see earlier comment about leveraging Resource's interfaces rather than defining the same thing here